sci-mathematics/mathematica/mathematica-13.0.1.ebuild

```diff
@@ -0,0 +1,140 @@
+# explicitly override with LD_PRELOAD.
+# RLink (libjri.so) requires dev-lang/R
+RDEPEND="
+	dev-lang/R
+	cuda? ( dev-util/nvidia-cuda-toolkit )
+	media-libs/freetype
+	virtual/libcrypt
+"
+
+DEPEND="
+	${RDEPEND}
+"
+
+BDEPEND="
+	dev-util/patchelf
+"
+
+# we need this a few times
+MPN="Mathematica"
+MPV=$(ver_cut 1-2)
+M_BINARIES="MathKernel Mathematica MathematicaScript WolframKernel WolframScript math mathematica mcc wolfram"
+M_TARGET="opt/Wolfram/${MPN}/${MPV}"
+
+# we might as well list all files in all QA variables...
+QA_PREBUILT="opt/*"
+
+S=${WORKDIR}
+
+src_unpack() {
+	/bin/sh "${DISTDIR}/${A}" --nox11 --keep --target "${S}/unpack" -- "-help" || die
+}
+
+src_prepare() {
+	pushd "${S}/unpack" > /dev/null || die
+	# fix ACCESS DENIED issue when installer check the avahi-daemon
+	sed -e "s|avahi-daemon -c|true|g" -i "Unix/Installer/MathInstaller" || die
+	/bin/sh "Unix/Installer/MathInstaller" -auto "-targetdir=${S}/${M_TARGET}" "-execdir=${S}/opt/bin" || die
+	popd > /dev/null || die
+	eapply_user
+}
+
+src_install() {
+	local ARCH='-x86-64'
+
+	if ! use doc; then
+		einfo "Removing documentation"
+		rm -r "${S}/${M_TARGET}/Documentation" || die
+	fi
+
+	einfo 'Removing MacOS- and Windows-specific files'
+	find "${S}/${M_TARGET}" -type d -\( -name Windows -o -name Windows-x86-64 \
+		-o -name MacOSX -o -name MacOSX-x86-64 -o -name Macintosh -\) \
+		-exec rm -rv {} + || die
+
+	if ! use cuda; then
+		einfo 'Removing cuda support'
+		rm -r "${S}/${M_TARGET}/SystemFiles/Components/CUDACompileTools/LibraryResources/Linux-x86-64/CUDAExtensions.so" || die
+	fi
+
+	# Linux-x86-64/AllVersions is the supported version, other versions remove
+	einfo 'Removing unsupported RLink versions'
+	rm -rf "${S}/${M_TARGET}/SystemFiles/Links/RLink/SystemFiles/Libraries/Linux-x86-64/3.5.0" || die
+	rm -rf "${S}/${M_TARGET}/SystemFiles/Links/RLink/SystemFiles/Libraries/Linux-x86-64/3.6.0" || die
+	rm -rf "${S}/${M_TARGET}/SystemFiles/Links/RLink/SystemFiles/Libraries/Linux/AllVersions" || die
+
+	# Fix RPATH
+	while IFS= read -r -d '' i; do
+		# Use \x7fELF header to separate ELF executables and libraries
+		# Skip .o files to avoid surprises
+		[[ -f "${i}" && "${i: -2}" != ".o" && $(od -t x1 -N 4 "${i}") == *"7f 45 4c 46"* ]] || continue
+		patchelf --set-rpath \
+'/'"${M_TARGET}"'/SystemFiles/Libraries/Linux-x86-64:'\
+'/'"${M_TARGET}"'/SystemFiles/Libraries/Linux-x86-64/Qt/lib:'\
+'/'"${M_TARGET}"'/SystemFiles/Java/Linux-x86-64/lib:'\
+'$ORIGIN' "${i}" || \
+		die "patchelf failed on ${i}"
+	done < <(find "${S}/${M_TARGET}" -type f -print0)
+
+	# move all over
+	mv "${S}"/opt "${D}"/opt || die
+
+	# the autogenerated symlinks point into sandbox, remove
+	rm "${D}"/opt/bin/* || die
+
+	# install wrappers instead
+	for name in ${M_BINARIES} ; do
+		einfo "Generating wrapper for ${name}"
+		echo '#!/bin/sh' >> "${T}/${name}" || die
+		echo "LD_PRELOAD=/usr/$(get_libdir)/libfreetype.so.6:/$(get_libdir)/libz.so.1 /${M_TARGET}/Executables/${name} \$*" \
```

Comment:
If I understand the comment in the ebuild correctly then if we add `virtual/libcrypt` as a dependency then we should also add libcrypt to `LD_PRELOAD` here, right?

Reply:
Now it is LD_PRELOAD=/usr/lib64/libfreetype.so.6:/lib64/libz.so.1:/lib64/libcrypt.so.1, after RPATH fix, this may fixed.